crates/bevy_utils/src/float_ord.rs

```diff
@@ -41,12 +41,12 @@ impl Hash for FloatOrd {
     fn hash<H: Hasher>(&self, state: &mut H) {
         if self.0.is_nan() {
             // Ensure all NaN representations hash to the same value
-            state.write(bytemuck::bytes_of(&f32::NAN));
+            state.write(&f32::to_ne_bytes(f32::NAN));
```

Comment:
What's up with this change?

Reply:
bevy_utils doesn't use bytemuck (currently, at least), so rather than add a bytemuck dep I just wrote the equivalent code with std support.

If it's preferred, we could just add bytemuck to bevy_utils, but it is in fact equivalent either way.

(It might be plausible that the bytemuck reëxport moves from bevy_core to bevy_utils as well, but I'm unsure about that one and just wanted to move `FloatOrd` here.)

Reply:
Sounds good to me; I'm content not to pull in the dependency here.